Add config flow to SMTP (#45553)
home-assistant/home-assistant.github.io: source/_integrations/smtp.markdown

Reviewer: ```suggestion
ha_domain: smtp
ha_config_flow: true
```

Reviewer: ```suggestion
ha_quality_scale: bronze
```

Author: the integration doesn't have a quality scale, so I assume i can just remove this

Reviewer: ```suggestion
ha_integration_type: service
```

    recipient: "YOUR_RECIPIENT"
```

Check your email provider configuration or help pages to get the correct SMTP settings. A restart of Home Assistant is required to pick up the configuration changes.

{% configuration %}
name:
  description: Setting the optional parameter `name` allows multiple notifiers to be created. The notifier will bind to the `notify.NOTIFIER_NAME` action.
  required: false
  type: string
  default: notify
sender:
  description: email address of the sender.
  required: true
  type: string
recipient:
  description: Default email address of the recipient of the notification. This can be a recipient address or a list of addresses for multiple recipients.<br>This is where you want to send your email notifications by default (when not specifying `target` in the action). Any email address(es) specified in the action's `target` field will override this recipient content.
  required: true
  type: [list, string]
server:
  description: SMTP server which is used to send the notifications.
  required: false
  type: string
  default: localhost
port:
  description: The port that the SMTP server is using.
  required: false
  type: integer
  default: 587

> timeout:

Reviewer: It seems there are less options available now. What happens if I have used timeout in YAML? Will ist still work?

Author: No, currently it will use the default value, I'm planning to add these settings in an options flow

>     description: "Hostname or IP address of the SMTP server."

Reviewer: ```suggestion
    description: "Hostname or IP address of your SMTP server. For example, `smtp.gmail.com`."
```
I think it would be helpful to have an example. Please change my example if you have a better one.

Author: I'd go for an RFC compliant example domain ^^ https://datatracker.ietf.org/doc/html/rfc2606#page-2

>     description: "SMTP server port number."

Reviewer: ```suggestion
    description: "Port number used by your SMTP server. Common values are `587` (STARTTLS) and `465` (TLS)."
```

>     description: "Encryption method used for the SMTP connection."

Reviewer: ```suggestion
    description: "Encryption method used for the SMTP connection. `starttls` upgrades a plain connection to encrypted (recommended). `tls` uses encryption from the start. `none` sends without encryption."
```
Would it be worth documenting the available options?

> 

Reviewer: ```suggestion

## Adding recipients

After the integration is set up, you need to add at least one recipient email address. Recipients are managed separately and can be added, edited, or removed at any time.

1. Go to {% my integrations title="**Settings** > **Devices & services**" %} and select the **SMTP** integration.
2. Select **Add recipient**.
3. Enter the email address you want to send notifications to.
4. Select **Submit**.

Repeat these steps to add more recipients. Every email address you add will receive notifications when you use the `notify.smtp` action.

```
would it be worth adding the steps how to add recipients?

Author: The first recipient is set up during the initial config flow, when the entry is created it automatically continues to the subentry flow. Also this integration doesn't create a `notify.smtp` action. I'll adjust it a little bit.

The **SMTP** {% term integration %} allows you to deliver notifications from Home Assistant to an email recipient.

{% include integrations/config_flow.md %}

Check your email provider configuration or help pages to get the correct SMTP settings.

> {% configuration %}

Reviewer: ```suggestion
{% configuration_basic %}
```
For the UI configuration description, we use the configuration_basic tag. https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L60

Sender email:
    description: "Email address that will appear in the From field."
    required: true

>     type: string

Reviewer: ```suggestion
```
configuration basic does not show required and type.

Sender name:
    description: "Display name shown as the email sender."
    required: false
    type: string
Host:
    description: "Hostname or IP address of the SMTP server."
    required: true
    type: string
Port:
    description: "SMTP server port number."
    required: true
    type: string
Connection security:
    description: "Encryption method used for the SMTP connection."
    required: true
    type: string
Username:
    description: "Username used to authenticate with the SMTP server."
    required: false
    type: string
Password:
    description: "Password or app-specific password for the SMTP account."
    required: false
    type: string
Verify SSL certificate:
    description: "Enable certificate verification for secure SSL/TLS connections."
    required: true
    type: string

> {% endconfiguration %}

Reviewer: ```suggestion
{% endconfiguration_basic %}
```
